src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/StashBox.cs

```diff
@@ -5,9 +5,16 @@
 
 namespace Microsoft.Extensions.DependencyInjection.Specification
 {
-    public class StashBoxDependencyInjectionSpecificationTests: DependencyInjectionSpecificationTests
+    public class StashBoxDependencyInjectionSpecificationTests : SkippableDependencyInjectionSpecificationTests
     {
-        protected override IServiceProvider CreateServiceProvider(IServiceCollection serviceCollection)
+        public override string[] SkippedTests => new[]
+        {
+            "PublicNoArgCtorConstrainedOpenGenericServicesCanBeResolved",
```

Comment:
Why are we skipping these tests? Same questions for the one in Autofac

Reply:
At one point, AutoFac didn't support constraints on `new`. I'm not sure that's relevant though, type constraints the useful bit for services. I'll double check though, it's been a few years since I tried this example when it failed.

The others were similar, they all worked with the desired scenarios, but not necessarily all constraints (like `struct`). I'll update and retry those.